packages/api/cms-api/src/generator/generate-crud.ts

```diff
@@ -549,9 +549,55 @@ ${
     `;
 }
 
+function generateRequiredPermissionDecorators({
+    generatorOptions,
+    metadata,
+}: {
+    generatorOptions: CrudGeneratorOptions;
+    metadata: EntityMetadata<any>;
+}) {
+    const { scopeProp } = buildOptions(metadata);
+    const ret: {
+        resolverDecorator?: string;
+        listDecorator?: string;
+        singleDecorator?: string;
+        createDecorator?: string;
+        updateDecorator?: string;
+        deleteDecorator?: string;
+    } = {};
+    if (generatorOptions.requiredPermission) {
+        const requiredPermission = generatorOptions.requiredPermission;
+        if (typeof requiredPermission === "string" || Array.isArray(requiredPermission)) {
+            ret.resolverDecorator = `@RequiredPermission(${JSON.stringify(
+                Array.isArray(requiredPermission) ? requiredPermission : [requiredPermission],
```

Comment:
this function became messy after checking for array. maybe it's alright to force the developer to use an array even for a single value. (as it's the same when using @RequiredPermission-decorator by hand)